File: src/game/ui/UserInterface.java
package game.ui;

import game.connection.RoomData;

/**
 * Created by neczp on 2016. 11. 05..
 */
public interface UserInterface {
	void connectionReady ();

	void connectionDropped ();

	void listRooms (RoomData[] roomData);

	void setIsInRoom (boolean isInRoom);

	void startGame (String mapName);

	void stopGame ();
}

File: iOS/InterstellarWar/InterstellarWar/Networking/UserInterface.swift
//
//  UserInterface.swift
//  InterstellarWar
//
//  Created by Neczpál Ábel on 2020. 02. 02..
//  Copyright © 2020. Neczpál Ábel. All rights reserved.
//

import Foundation

protocol UserInterface {
    
    func connectionReady ();

    func connectionDropped ();

    func listRooms (_ roomData : [JSON]);

    func setIsInRoom (_ isInRoom : Bool);

    func startGame (_ mapName : String);

    func stopGame ();
}
